src/contexts/UserProvider.tsx

```diff
@@ -1,7 +1,8 @@
-import { createContext, useContext, useEffect, useState } from "react";
+import React, { createContext, useContext, useEffect, useState } from "react";
```

Comment:
are you sure you need to import React here? No unused imports! (maybe you do need it for some reason though I'm just asking)

Reply:
yes it is being used for typing